packages/lesswrong/lib/sql/PgCollection.ts

```diff
@@ -0,0 +1,245 @@
+ * to instead implement CollectionBase.
+ */
+class PgCollection<T extends DbObject> extends MongoCollection<T> {
+  table: Table;
+
+  constructor(tableName: string, options?: { _suppressSameNameError?: boolean }) {
+    super(tableName, options);
+  }
+
+  isPostgres() {
+    return true;
+  }
+
+  isConnected() {
+    return !!getSqlClient();
+  }
+
+  buildPostgresTable() {
+    this.table = Table.fromCollection(this as unknown as CollectionBase<T>);
+  }
+
+  /**
+   * Execute the given query
+   * The `debugData` parameter is completely optional and is only used to improve
+   * the error message if something goes wrong
+   */
+  async executeQuery(query: Query<T>, debugData?: any): Promise<any[]> {
+    executingQueries++;
+    let result: any[];
+    try {
+      const {sql, args} = query.compile();
+      const client = getSqlClientOrThrow();
+      result = await client.any(sql, args);
+    } catch (error) {
+      // If this error gets triggered, you probably generated a malformed query
+      const {collectionName} = this as unknown as CollectionBase<T>;
+      debugData = util.inspect({collectionName, ...debugData}, {depth: null});
+      const {sql, args} = query.compile();
+      // eslint-disable-next-line no-console
+      console.error(`SQL Error: ${error.message}: \`${sql}\`: ${util.inspect(args)}: ${debugData}`);
+      throw error;
+    } finally {
+      executingQueries--;
+    }
+    return result;
+  }
+
+  getTable = () => {
+    if (bundleIsServer) {
+      return this.table;
+    } else {
+      throw new Error("Attempted to run postgres query on the client");
+    }
+  }
+
+  find = (selector?: MongoSelector<T>, options?: MongoFindOptions<T>): FindResult<T> => {
+    return {
+      fetch: async () => {
+        const select = new SelectQuery<T>(this.getTable(), selector, options);
+        const result = await this.executeQuery(select, {selector, options});
+        return result as unknown as T[];
+      },
+      count: async () => {
+        const select = new SelectQuery(this.getTable(), selector, options, {count: true});
+        const result = await this.executeQuery(select, {selector, options});
+        return result?.[0].count ?? 0;
+      },
+    };
+  }
+
+  findOne = async (
+    selector?: string | MongoSelector<T>,
+    options?: MongoFindOneOptions<T>,
+    projection?: MongoProjection<T>,
+  ): Promise<T|null> => {
+    const select = new SelectQuery<T>(this.getTable(), selector, {limit: 1, ...options, projection});
+    const result = await this.executeQuery(select, {selector, options, projection});
+    return result ? result[0] as unknown as T : null;
+  }
+
+  findOneArbitrary = async (): Promise<T|null> => {
+    const select = new SelectQuery<T>(this.getTable(), undefined, {limit: 1});
+    const result = await this.executeQuery(select);
+    return result ? result[0] as unknown as T : null;
+  }
+
+  rawInsert = async (data: T, options: MongoInsertOptions<T>) => {
```

Comment:
nit: add some kind of typing at some stage here
```suggestion
  rawInsert = async (data: T, options: MongoInsertOptions<T>): Promise<string> => {
```
also can't you insert multiple rows at once, so this should be able to return an array of ids?